### ISSUE TITLE
Hard crash when combined CSS transform is not an identity or transform

### ISSUE THREAD

**Issue by yyny** [LadybirdBrowser/ladybird#2055]
This code

https://github.com/LadybirdBrowser/ladybird/blob/3180df3337cc319cdc30b33b70a19b3aafa5642b/Userland/Libraries/LibWeb/Painting/StackingContext.cpp#L323-L330

assumes that the `masking_area` has a value if the `clip_path` has a value, and is a basic shape.

However, here:

https://github.com/LadybirdBrowser/ladybird/blob/3180df3337cc319cdc30b33b70a19b3aafa5642b/Userland/Libraries/LibWeb/Painting/PaintableBox.cpp#L995-L997

`masking_area` is left undefined if the combined CSS transform is not an identity or transform, resulting in a hard crash.

This situation could probably be handled more gracefully.

**Comment by AtkinsSJ** [comment]
Thanks, do you have a small test case for this?

**Comment by yyny** [comment]
Unfortunately not, it took a lot of clicking around on various websites.

**Comment by shlyakpavel** [comment]
Try
```html
<!DOCTYPE html>
<html>
<head>
<style>
.animated-box {
    width: 100px;
    height: 100px;
    background: linear-gradient(to right, red, blue);
    clip-path: polygon(50% 0%, 100% 100%, 0% 100%);
    animation: rotateScale 5s infinite;
}

@keyframes rotateScale {
    0% {
        transform: rotate(0deg) scale(1);
    }
    50% {
        transform: rotate(180deg) scale(2);
    }
    100% {
        transform: rotate(360deg) scale(1);
    }
}
</style>
</head>
<body>
<div class="animated-box"></div>
</body>
</html>
```

**Comment by yyny** [comment]
Fixed in #2095
